### ISSUE TITLE
Double control C to abort a scan

### ISSUE THREAD

**Issue by ambarb** [NSLS-II/bluesky#226]
Aborted the scan using by hitting control C twice.  Opened new figure. Could not launch scan.  gs.RE.abort did not fix. Close plotting windows (generates error report).  Scan would still not launch.  Restart bluesky works.

See below.

```python
In [194]: grabit
Select area of screen to grab .........

In [195]: ascan(epu2_gap,21,22,20)
Transient Scan ID: 52140
Persistent Unique Scan ID: 'a502454f-8f5f-48a6-89bb-1a084522d081'
+------------+--------------+----------------+----------------+
|   seq_num  |        time  |      epu2_gap  |    sclr_chan2  |
+------------+--------------+----------------+----------------+
^C
^C+------------+--------------+----------------+----------------+
/opt/conda_envs/collection/lib/python3.4/site-packages/numpy/core/_methods.py:59: RuntimeWarning: Mean of empty slice.
  warnings.warn("Mean of empty slice.", RuntimeWarning)
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
/opt/conda_envs/collection/lib/python3.4/site-packages/requests/packages/urllib3/connectionpool.py in _make_request(self, conn, method, url, timeout, **httplib_request_kw)
    371             try:  # Python 2.7, use buffering of HTTP responses
--> 372                 httplib_response = conn.getresponse(buffering=True)
    373             except TypeError:  # Python 2.6 and older

TypeError: getresponse() got an unexpected keyword argument 'buffering'

During handling of the above exception, another exception occurred:

KeyboardInterrupt                         Traceback (most recent call last)
<ipython-input-195-e9cbc2869e74> in <module>()
----> 1 ascan(epu2_gap,21,22,20)

/opt/conda_envs/collection/lib/python3.4/site-packages/bluesky/simple_scans.py in __call__(self, motor, start, finish, intervals, time, subs, **kwargs)
    209         original_times = _set_acquire_time(time)
    210         result = super().__call__(motor, start, finish, intervals + 1,
--> 211                                   subs=subs, **kwargs)
    212         _unset_acquire_time(original_times)
    213         return result

/opt/conda_envs/collection/lib/python3.4/site-packages/bluesky/simple_scans.py in __call__(self, subs, sub_factories, *args, **kwargs)
    136         # Any remainging kwargs go the RE. To be safe, no args are passed
    137         # to RE; RE args effectively become keyword-only arguments.
--> 138         return gs.RE(self.scan, _subs, **kwargs)
    139 
    140 

/opt/conda_envs/collection/lib/python3.4/site-packages/bluesky/run_engine.py in __call__(self, plan, subs, **metadata)
    512         with SignalHandler(signal.SIGINT) as self._sigint_handler:  # ^C
    513             self._task = loop.create_task(self._run())
--> 514             loop.run_forever()
    515             if self._task.done() and not self._task.cancelled():
    516                 exc = self._task.exception()

/opt/conda_envs/collection/lib/python3.4/asyncio/base_events.py in run_forever(self)
    274             while True:
    275                 try:
--> 276                     self._run_once()
    277                 except _StopError:
    278                     break

/opt/conda_envs/collection/lib/python3.4/asyncio/base_events.py in _run_once(self)
   1170                     self._current_handle = None
   1171             else:
-> 1172                 handle._run()
   1173         handle = None  # Needed to break cycles when an exception occurs.
   1174 

/opt/conda_envs/collection/lib/python3.4/asyncio/events.py in _run(self)
    118     def _run(self):
    119         try:
--> 120             self._callback(*self._args)
    121         except Exception as exc:
    122             cb = _format_callback(self._callback, self._args)

/opt/conda_envs/collection/lib/python3.4/asyncio/tasks.py in _wakeup(self, future)
    290             self._step(None, exc)
    291         else:
--> 292             self._step(value, None)
    293         self = None  # Needed to break cycles when an exception occurs.
    294 

/opt/conda_envs/collection/lib/python3.4/asyncio/tasks.py in _step(***failed resolving arguments***)
    236                 result = coro.send(value)
    237             else:
--> 238                 result = next(coro)
    239         except StopIteration as exc:
    240             self.set_result(exc.value)

/opt/conda_envs/collection/lib/python3.4/site-packages/bluesky/run_engine.py in _run(self)
    640                 self.debug("About to process: {0}, {1}".format(coro, msg))
    641                 yield from asyncio.sleep(0.001)  # TODO Do we need this?
--> 642                 response = yield from coro(msg)
    643                 self.debug('RE.state: ' + self.state)
    644                 self.debug('msg: {}\n   response: {}'.format(msg, response))

/opt/conda_envs/collection/lib/python3.4/asyncio/coroutines.py in coro(*args, **kw)
    139         @functools.wraps(func)
    140         def coro(*args, **kw):
--> 141             res = func(*args, **kw)
    142             if isinstance(res, futures.Future) or inspect.isgenerator(res):
    143                 res = yield from res

/opt/conda_envs/collection/lib/python3.4/site-packages/bluesky/run_engine.py in _logbook(self, msg)
   1001             d.update(self._metadata_per_run)
   1002             d.update(msg.kwargs)
-> 1003             return self.logbook(log_message, d)
   1004 
   1005     @asyncio.coroutine

/opt/conda_envs/collection/lib/python3.4/site-packages/bluesky/standard_config.py in _logbook_log(msg, d)
     88                     # properties={'Bluesky': d},
     89                     ensure=True,
---> 90                     logbooks=logbooks)
     91 
     92     return _logbook_log

/opt/conda_envs/collection/lib/python3.4/site-packages/pyOlog-0.3.0-py3.4.egg/pyOlog/SimpleOlogClient.py in log(self, text, logbooks, tags, properties, attachments, verify, ensure)
    277         if logbooks:
    278             for x in logbooks:
--> 279                 if x not in self.logbooks:
    280                     if ensure:
    281                         self.create_logbook(x)

/opt/conda_envs/collection/lib/python3.4/site-packages/pyOlog-0.3.0-py3.4.egg/pyOlog/SimpleOlogClient.py in logbooks(self)
    101             Logbook names as string
    102         """
--> 103         return [l.name for l in self.session.list_logbooks()]
    104 
    105     @property

/opt/conda_envs/collection/lib/python3.4/site-packages/pyOlog-0.3.0-py3.4.egg/pyOlog/OlogClient.py in list_logbooks(self)
    244         List all logbooks in the Olog.
    245         '''
--> 246         resp = self._get(self.logbooks_resource)
    247 
    248         logbooks = []

/opt/conda_envs/collection/lib/python3.4/site-packages/pyOlog-0.3.0-py3.4.egg/pyOlog/OlogClient.py in _get(self, url, **kwargs)
     99         logger.debug("HTTP GET to %s", self._url + url)
    100         kwargs.update({'headers': self.json_header})
--> 101         resp = self._session.get(self._url + url, **kwargs)
    102         resp.raise_for_status()
    103         return resp

/opt/conda_envs/collection/lib/python3.4/site-packages/requests/sessions.py in get(self, url, **kwargs)
    475 
    476         kwargs.setdefault('allow_redirects', True)
--> 477         return self.request('GET', url, **kwargs)
    478 
    479     def options(self, url, **kwargs):

/opt/conda_envs/collection/lib/python3.4/site-packages/requests/sessions.py in request(self, method, url, params, data, headers, cookies, files, auth, timeout, allow_redirects, proxies, hooks, stream, verify, cert, json)
    463         }
    464         send_kwargs.update(settings)
--> 465         resp = self.send(prep, **send_kwargs)
    466 
    467         return resp

/opt/conda_envs/collection/lib/python3.4/site-packages/requests/sessions.py in send(self, request, **kwargs)
    571 
    572         # Send the request
--> 573         r = adapter.send(request, **kwargs)
    574 
    575         # Total elapsed time of the request (approximately)

/opt/conda_envs/collection/lib/python3.4/site-packages/requests/adapters.py in send(self, request, stream, timeout, verify, cert, proxies)
    368                     decode_content=False,
    369                     retries=self.max_retries,
--> 370                     timeout=timeout
    371                 )
    372 

/opt/conda_envs/collection/lib/python3.4/site-packages/requests/packages/urllib3/connectionpool.py in urlopen(self, method, url, body, headers, retries, redirect, assert_same_host, timeout, pool_timeout, release_conn, **response_kw)
    542             httplib_response = self._make_request(conn, method, url,
    543                                                   timeout=timeout_obj,
--> 544                                                   body=body, headers=headers)
    545 
    546             # If we're going to release the connection in ``finally:``, then

/opt/conda_envs/collection/lib/python3.4/site-packages/requests/packages/urllib3/connectionpool.py in _make_request(self, conn, method, url, timeout, **httplib_request_kw)
    372                 httplib_response = conn.getresponse(buffering=True)
    373             except TypeError:  # Python 2.6 and older
--> 374                 httplib_response = conn.getresponse()
    375         except (SocketTimeout, BaseSSLError, SocketError) as e:
    376             self._raise_timeout(err=e, url=url, timeout_value=read_timeout)

/opt/conda_envs/collection/lib/python3.4/http/client.py in getresponse(self)
   1169 
   1170         try:
-> 1171             response.begin()
   1172             assert response.will_close != _UNKNOWN
   1173             self.__state = _CS_IDLE

/opt/conda_envs/collection/lib/python3.4/http/client.py in begin(self)
    349         # read until we get a non-100 response
    350         while True:
--> 351             version, status, reason = self._read_status()
    352             if status != CONTINUE:
    353                 break

/opt/conda_envs/collection/lib/python3.4/http/client.py in _read_status(self)
    311 
    312     def _read_status(self):
--> 313         line = str(self.fp.readline(_MAXLINE + 1), "iso-8859-1")
    314         if len(line) > _MAXLINE:
    315             raise LineTooLong("status line")

/opt/conda_envs/collection/lib/python3.4/socket.py in readinto(self, b)
    372         while True:
    373             try:
--> 374                 return self._sock.recv_into(b)
    375             except timeout:
    376                 self._timeout_occurred = True

/opt/conda_envs/collection/lib/python3.4/ssl.py in recv_into(self, buffer, nbytes, flags)
    749                   "non-zero flags not allowed in calls to recv_into() on %s" %
    750                   self.__class__)
--> 751             return self.read(nbytes, buffer)
    752         else:
    753             return socket.recv_into(self, buffer, nbytes, flags)

/opt/conda_envs/collection/lib/python3.4/ssl.py in read(self, len, buffer)
    621         try:
    622             if buffer is not None:
--> 623                 v = self._sslobj.read(len, buffer)
    624             else:
    625                 v = self._sslobj.read(len or 1024)

KeyboardInterrupt: 

In [196]: figure
--------> figure()
Out[196]: <matplotlib.figure.Figure at 0x7eff942278d0>

In [197]: ascan(epu2_gap,21,22,20)
Out[197]: []

In [198]: 
KeyboardInterrupt

In [198]: abort()
Aborting....

In [199]: ascan(epu2_gap,21,22,20)
---------------------------------------------------------------------------
RuntimeError                              Traceback (most recent call last)
<ipython-input-199-e9cbc2869e74> in <module>()
----> 1 ascan(epu2_gap,21,22,20)

/opt/conda_envs/collection/lib/python3.4/site-packages/bluesky/simple_scans.py in __call__(self, motor, start, finish, intervals, time, subs, **kwargs)
    209         original_times = _set_acquire_time(time)
    210         result = super().__call__(motor, start, finish, intervals + 1,
--> 211                                   subs=subs, **kwargs)
    212         _unset_acquire_time(original_times)
    213         return result

/opt/conda_envs/collection/lib/python3.4/site-packages/bluesky/simple_scans.py in __call__(self, subs, sub_factories, *args, **kwargs)
    136         # Any remainging kwargs go the RE. To be safe, no args are passed
    137         # to RE; RE args effectively become keyword-only arguments.
--> 138         return gs.RE(self.scan, _subs, **kwargs)
    139 
    140 

/opt/conda_envs/collection/lib/python3.4/site-packages/bluesky/run_engine.py in __call__(self, plan, subs, **metadata)
    459         # First thing's first: if we are in the wrong state, raise.
    460         if not self.state.is_idle:
--> 461             raise RuntimeError("The RunEngine is in a %s state" % self.state)
    462         if self._panic:
    463             raise PanicError("RunEngine is panicked. The run "

RuntimeError: The RunEngine is in a running state

In [200]: 
KeyboardInterrupt

In [200]: abort()
Aborting....

In [201]: ascan(epu2_gap,21,22,20)
---------------------------------------------------------------------------
RuntimeError                              Traceback (most recent call last)
<ipython-input-201-e9cbc2869e74> in <module>()
----> 1 ascan(epu2_gap,21,22,20)

/opt/conda_envs/collection/lib/python3.4/site-packages/bluesky/simple_scans.py in __call__(self, motor, start, finish, intervals, time, subs, **kwargs)
    209         original_times = _set_acquire_time(time)
    210         result = super().__call__(motor, start, finish, intervals + 1,
--> 211                                   subs=subs, **kwargs)
    212         _unset_acquire_time(original_times)
    213         return result

/opt/conda_envs/collection/lib/python3.4/site-packages/bluesky/simple_scans.py in __call__(self, subs, sub_factories, *args, **kwargs)
    136         # Any remainging kwargs go the RE. To be safe, no args are passed
    137         # to RE; RE args effectively become keyword-only arguments.
--> 138         return gs.RE(self.scan, _subs, **kwargs)
    139 
    140 

/opt/conda_envs/collection/lib/python3.4/site-packages/bluesky/run_engine.py in __call__(self, plan, subs, **metadata)
    459         # First thing's first: if we are in the wrong state, raise.
    460         if not self.state.is_idle:
--> 461             raise RuntimeError("The RunEngine is in a %s state" % self.state)
    462         if self._panic:
    463             raise PanicError("RunEngine is panicked. The run "

RuntimeError: The RunEngine is in a running state

In [202]: gs.RE.a
gs.RE.abort        gs.RE.all_is_well  

In [202]: gs.RE.abort
--------> gs.RE.abort()
Aborting....

In [203]: ascan(epu2_gap,21,22,20)
---------------------------------------------------------------------------
RuntimeError                              Traceback (most recent call last)
<ipython-input-203-e9cbc2869e74> in <module>()
----> 1 ascan(epu2_gap,21,22,20)

/opt/conda_envs/collection/lib/python3.4/site-packages/bluesky/simple_scans.py in __call__(self, motor, start, finish, intervals, time, subs, **kwargs)
    209         original_times = _set_acquire_time(time)
    210         result = super().__call__(motor, start, finish, intervals + 1,
--> 211                                   subs=subs, **kwargs)
    212         _unset_acquire_time(original_times)
    213         return result

/opt/conda_envs/collection/lib/python3.4/site-packages/bluesky/simple_scans.py in __call__(self, subs, sub_factories, *args, **kwargs)
    136         # Any remainging kwargs go the RE. To be safe, no args are passed
    137         # to RE; RE args effectively become keyword-only arguments.
--> 138         return gs.RE(self.scan, _subs, **kwargs)
    139 
    140 

/opt/conda_envs/collection/lib/python3.4/site-packages/bluesky/run_engine.py in __call__(self, plan, subs, **metadata)
    459         # First thing's first: if we are in the wrong state, raise.
    460         if not self.state.is_idle:
--> 461             raise RuntimeError("The RunEngine is in a %s state" % self.state)
    462         if self._panic:
    463             raise PanicError("RunEngine is panicked. The run "

RuntimeError: The RunEngine is in a running state

In [204]: gs.PLOT_Y ='sclr_chan2'

In [205]: gs.TABLE_COLS =['sclr_chan2']

In [206]: gs.COUNT_TIME=1

In [207]: ascan(epu2_gap,21,22,20)
---------------------------------------------------------------------------
RuntimeError                              Traceback (most recent call last)
<ipython-input-207-e9cbc2869e74> in <module>()
----> 1 ascan(epu2_gap,21,22,20)

/opt/conda_envs/collection/lib/python3.4/site-packages/bluesky/simple_scans.py in __call__(self, motor, start, finish, intervals, time, subs, **kwargs)
    209         original_times = _set_acquire_time(time)
    210         result = super().__call__(motor, start, finish, intervals + 1,
--> 211                                   subs=subs, **kwargs)
    212         _unset_acquire_time(original_times)
    213         return result

/opt/conda_envs/collection/lib/python3.4/site-packages/bluesky/simple_scans.py in __call__(self, subs, sub_factories, *args, **kwargs)
    136         # Any remainging kwargs go the RE. To be safe, no args are passed
    137         # to RE; RE args effectively become keyword-only arguments.
--> 138         return gs.RE(self.scan, _subs, **kwargs)
    139 
    140 

/opt/conda_envs/collection/lib/python3.4/site-packages/bluesky/run_engine.py in __call__(self, plan, subs, **metadata)
    459         # First thing's first: if we are in the wrong state, raise.
    460         if not self.state.is_idle:
--> 461             raise RuntimeError("The RunEngine is in a %s state" % self.state)
    462         if self._panic:
    463             raise PanicError("RunEngine is panicked. The run "

RuntimeError: The RunEngine is in a running state

In [208]: figure
--------> figure()
Out[208]: <matplotlib.figure.Figure at 0x7eff83f98ac8>

In [209]: gs.RE.abort
--------> gs.RE.abort()
Aborting....

In [210]: gs.TABLE_COLS =['sclr_chan2']

In [211]: gs.PLOT_Y ='sclr_chan2'

In [212]: gs.COUNT_TIME=1

In [213]: gs.PLOT_Y ='sclr_chan2'Task exception was never retrieved
future: <Task cancelling coro=<_run() done, defined at /opt/conda_envs/collection/lib/python3.4/site-packages/bluesky/run_engine.py:614> exception=KeyboardInterrupt()>
Traceback (most recent call last):
  File "/opt/conda_envs/collection/lib/python3.4/site-packages/requests/packages/urllib3/connectionpool.py", line 372, in _make_request
    httplib_response = conn.getresponse(buffering=True)
TypeError: getresponse() got an unexpected keyword argument 'buffering'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/opt/conda_envs/collection/lib/python3.4/site-packages/IPython/core/interactiveshell.py", line 3066, in run_code
    exec(code_obj, self.user_global_ns, self.user_ns)
  File "<ipython-input-195-e9cbc2869e74>", line 1, in <module>
    ascan(epu2_gap,21,22,20)
  File "/opt/conda_envs/collection/lib/python3.4/site-packages/bluesky/simple_scans.py", line 211, in __call__
    subs=subs, **kwargs)
  File "/opt/conda_envs/collection/lib/python3.4/site-packages/bluesky/simple_scans.py", line 138, in __call__
    return gs.RE(self.scan, _subs, **kwargs)
  File "/opt/conda_envs/collection/lib/python3.4/site-packages/bluesky/run_engine.py", line 514, in __call__
    loop.run_forever()
  File "/opt/conda_envs/collection/lib/python3.4/asyncio/base_events.py", line 276, in run_forever
    self._run_once()
  File "/opt/conda_envs/collection/lib/python3.4/asyncio/base_events.py", line 1172, in _run_once
    handle._run()
  File "/opt/conda_envs/collection/lib/python3.4/asyncio/events.py", line 120, in _run
    self._callback(*self._args)
  File "/opt/conda_envs/collection/lib/python3.4/asyncio/tasks.py", line 292, in _wakeup
    self._step(value, None)
  File "/opt/conda_envs/collection/lib/python3.4/asyncio/tasks.py", line 238, in _step
    result = next(coro)
  File "/opt/conda_envs/collection/lib/python3.4/site-packages/bluesky/run_engine.py", line 642, in _run
    response = yield from coro(msg)
  File "/opt/conda_envs/collection/lib/python3.4/asyncio/coroutines.py", line 141, in coro
    res = func(*args, **kw)
  File "/opt/conda_envs/collection/lib/python3.4/site-packages/bluesky/run_engine.py", line 1003, in _logbook
    return self.logbook(log_message, d)
  File "/opt/conda_envs/collection/lib/python3.4/site-packages/bluesky/standard_config.py", line 90, in _logbook_log
    logbooks=logbooks)
  File "/opt/conda_envs/collection/lib/python3.4/site-packages/pyOlog-0.3.0-py3.4.egg/pyOlog/SimpleOlogClient.py", line 279, in log
    if x not in self.logbooks:
  File "/opt/conda_envs/collection/lib/python3.4/site-packages/pyOlog-0.3.0-py3.4.egg/pyOlog/SimpleOlogClient.py", line 103, in logbooks
    return [l.name for l in self.session.list_logbooks()]
  File "/opt/conda_envs/collection/lib/python3.4/site-packages/pyOlog-0.3.0-py3.4.egg/pyOlog/OlogClient.py", line 246, in list_logbooks
    resp = self._get(self.logbooks_resource)
  File "/opt/conda_envs/collection/lib/python3.4/site-packages/pyOlog-0.3.0-py3.4.egg/pyOlog/OlogClient.py", line 101, in _get
    resp = self._session.get(self._url + url, **kwargs)
  File "/opt/conda_envs/collection/lib/python3.4/site-packages/requests/sessions.py", line 477, in get
    return self.request('GET', url, **kwargs)
  File "/opt/conda_envs/collection/lib/python3.4/site-packages/requests/sessions.py", line 465, in request
    resp = self.send(prep, **send_kwargs)
  File "/opt/conda_envs/collection/lib/python3.4/site-packages/requests/sessions.py", line 573, in send
    r = adapter.send(request, **kwargs)
  File "/opt/conda_envs/collection/lib/python3.4/site-packages/requests/adapters.py", line 370, in send
    timeout=timeout
  File "/opt/conda_envs/collection/lib/python3.4/site-packages/requests/packages/urllib3/connectionpool.py", line 544, in urlopen
    body=body, headers=headers)
  File "/opt/conda_envs/collection/lib/python3.4/site-packages/requests/packages/urllib3/connectionpool.py", line 374, in _make_request
    httplib_response = conn.getresponse()
  File "/opt/conda_envs/collection/lib/python3.4/http/client.py", line 1171, in getresponse
    response.begin()
  File "/opt/conda_envs/collection/lib/python3.4/http/client.py", line 351, in begin
    version, status, reason = self._read_status()
  File "/opt/conda_envs/collection/lib/python3.4/http/client.py", line 313, in _read_status
    line = str(self.fp.readline(_MAXLINE + 1), "iso-8859-1")
  File "/opt/conda_envs/collection/lib/python3.4/socket.py", line 374, in readinto
    return self._sock.recv_into(b)
  File "/opt/conda_envs/collection/lib/python3.4/ssl.py", line 751, in recv_into
    return self.read(nbytes, buffer)
  File "/opt/conda_envs/collection/lib/python3.4/ssl.py", line 623, in read
    v = self._sslobj.read(len, buffer)
KeyboardInterrupt
/opt/conda_envs/collection/lib/python3.4/site-packages/matplotlib/_pylab_helpers.py:72: ResourceWarning: unclosed <ssl.SSLSocket fd=20, family=AddressFamily.AF_INET, type=SocketKind.SOCK_STREAM, proto=6, laddr=('10.23.0.209', 51877), raddr=('10.23.0.8', 8181)>
  gc.collect(1)
             COUNT_TIME=1

In [214]: abort()
Aborting....

In [215]: figure
--------> figure()
Out[215]: <matplotlib.figure.Figure at 0x7eff83fc1780>

In [216]: ascan(epu2_gap,21,22,20)
---------------------------------------------------------------------------
RuntimeError                              Traceback (most recent call last)
<ipython-input-216-e9cbc2869e74> in <module>()
----> 1 ascan(epu2_gap,21,22,20)

/opt/conda_envs/collection/lib/python3.4/site-packages/bluesky/simple_scans.py in __call__(self, motor, start, finish, intervals, time, subs, **kwargs)
    209         original_times = _set_acquire_time(time)
    210         result = super().__call__(motor, start, finish, intervals + 1,
--> 211                                   subs=subs, **kwargs)
    212         _unset_acquire_time(original_times)
    213         return result

/opt/conda_envs/collection/lib/python3.4/site-packages/bluesky/simple_scans.py in __call__(self, subs, sub_factories, *args, **kwargs)
    136         # Any remainging kwargs go the RE. To be safe, no args are passed
    137         # to RE; RE args effectively become keyword-only arguments.
--> 138         return gs.RE(self.scan, _subs, **kwargs)
    139 
    140 

/opt/conda_envs/collection/lib/python3.4/site-packages/bluesky/run_engine.py in __call__(self, plan, subs, **metadata)
    459         # First thing's first: if we are in the wrong state, raise.
    460         if not self.state.is_idle:
--> 461             raise RuntimeError("The RunEngine is in a %s state" % self.state)
    462         if self._panic:
    463             raise PanicError("RunEngine is panicked. The run "

RuntimeError: The RunEngine is in a running state

In [217]: abort()
Aborting....

In [218]: gs.RE.abort
--------> gs.RE.abort()
Aborting....

In [219]: gs.COUNT_TIME=1

In [220]: gs.TABLE_COLS =['sclr_chan2']

In [221]: gs.PLOT_Y ='sclr_chan2'

In [222]: gs.TABLE_COLS =['sclr_chan2']

In [223]: ascan(epu2_gap,21,22,20)
---------------------------------------------------------------------------
RuntimeError                              Traceback (most recent call last)
<ipython-input-223-e9cbc2869e74> in <module>()
----> 1 ascan(epu2_gap,21,22,20)

/opt/conda_envs/collection/lib/python3.4/site-packages/bluesky/simple_scans.py in __call__(self, motor, start, finish, intervals, time, subs, **kwargs)
    209         original_times = _set_acquire_time(time)
    210         result = super().__call__(motor, start, finish, intervals + 1,
--> 211                                   subs=subs, **kwargs)
    212         _unset_acquire_time(original_times)
    213         return result

/opt/conda_envs/collection/lib/python3.4/site-packages/bluesky/simple_scans.py in __call__(self, subs, sub_factories, *args, **kwargs)
    136         # Any remainging kwargs go the RE. To be safe, no args are passed
    137         # to RE; RE args effectively become keyword-only arguments.
--> 138         return gs.RE(self.scan, _subs, **kwargs)
    139 
    140 

/opt/conda_envs/collection/lib/python3.4/site-packages/bluesky/run_engine.py in __call__(self, plan, subs, **metadata)
    459         # First thing's first: if we are in the wrong state, raise.
    460         if not self.state.is_idle:
--> 461             raise RuntimeError("The RunEngine is in a %s state" % self.state)
    462         if self._panic:
    463             raise PanicError("RunEngine is panicked. The run "

RuntimeError: The RunEngine is in a running state

In [224]: 
KeyboardInterrupt

In [224]: eixt
---------------------------------------------------------------------------
NameError                                 Traceback (most recent call last)
<ipython-input-224-e8609674cbb9> in <module>()
----> 1 eixt

NameError: name 'eixt' is not defined

In [225]: exit
/opt/conda_envs/collection/lib/python3.4/site-packages/path.py:1715: DeprecationWarning: path is deprecated. Use Path instead.
  warnings.warn(msg, DeprecationWarning)
Stored '_persisting' (list)
Stored '_scan_id' (int)
/opt/conda_envs/collection/lib/python3.4/asyncio/base_events.py:359: ResourceWarning: unclosed event loop <_UnixSelectorEventLoop running=False closed=False debug=False>
/opt/conda_envs/collection/lib/python3.4/importlib/_bootstrap.py:2150: ImportWarning: sys.meta_path is empty
Task was destroyed but it is pending!
task: <Task cancelling coro=<_run() running at /opt/conda_envs/collection/lib/python3.4/site-packages/bluesky/run_engine.py:614>>
sys:1: ResourceWarning: unclosed file <_io.TextIOWrapper name='/dev/null' mode='w' encoding='UTF-8'>
(collection)[xf23id1@xf23id-ws3 ~]$ ipython --profile=collection
Activating auto-logging. Current session state plus future input saved.
Filename       : /epics/xf/23id/ophyd/logs/xf23id1_log.py
Mode           : append
Output logging : False
Raw input log  : False
Timestamping   : False
State          : active
Python 3.4.3 |Continuum Analytics, Inc.| (default, Jun  4 2015, 15:29:08) 
Type "copyright", "credits" or "license" for more information.

IPython 4.0.0 -- An enhanced Interactive Python.
?         -> Introduction and overview of IPython's features.
%quickref -> Quick reference.
help      -> Python's own help system.
object?   -> Details about 'object', use 'object??' for extra details.

IPython profile: collection
Gtk-Message: Failed to load module "canberra-gtk-module"
Using matplotlib backend: Qt4Agg
Loading Ophyd Session Manager...
2015-10-09 13:18:08,354 [ophyd_session:DEBUG] Last persistence list = ['_persisting', '_scan_id']
/opt/conda_envs/collection/lib/python3.4/site-packages/path.py:1715: DeprecationWarning: path is deprecated. Use Path instead.
  warnings.warn(msg, DeprecationWarning)
Stored '_persisting' (list)
2015-10-09 13:18:08,361 [ophyd_session:DEBUG] Last Scan ID = 10657
Stored '_scan_id' (int)
...Done loading Ophyd Session Manager
/opt/conda_envs/collection/lib/python3.4/site-packages/pyOlog-0.3.0-py3.4.egg/pyOlog/conf.py:37: DeprecationWarning: The SafeConfigParser class has been renamed to ConfigParser in Python 3.2. This alias will be removed in future versions. Use ConfigParser directly instead.
Found metadata history in existing file.
```

**Comment by danielballan** [comment]
I saw something like this once before but could not reproduce it. Apparently the abort method is silently failing to do its job. We need to find a way to make it more robust.

**Comment by ericdill** [comment]
So just sat with @ambarb at CSX for a while trying to reproduce this error.  We have concluded that the windows solution is fine with @ambarb .

The windows solution is "turn it off and back on again" :smiling_imp: 

Having just read through the stack trace carefully, it seems that @ambarb managed to hit ctrl+c in the middle of the attempt to write to the OLog!  I didn't even know that was possible.  As such, it would be magnificently difficult to reproduce such an error.

**Comment by danielballan** [comment]
Hmm. That still doesn't explain why her repeated attempts to abort failed silently.

**Comment by danielballan** [comment]
I think the issue with abort being effectively ignored was resolved by #231. I'll leave this open and see how reliably `abort` is behaving in a week or two.

**Comment by danielballan** [comment]
Looks good so far. @ambarb please reopen if you have this issue again.
